pkg/envinfo/url.go

```diff
@@ -85,13 +101,19 @@ func (ei *EnvInfo) CompleteURL(url *localConfigProvider.LocalURL) error {
 		if err != nil {
 			return err
 		}
-		portsOf = fmt.Sprintf("component %s", ei.GetName())
+		err = ei.completeNoPortSpecified(url, fmt.Sprintf("component %s", ei.GetName()), ports)
+		if err != nil {
+			return err
+		}
 	} else {
 		ports, err = ei.GetContainerPorts(url.Container)
 		if err != nil {
 			return err
 		}
-		portsOf = fmt.Sprintf("container %s", url.Container)
+		err = ei.completeNoPortSpecified(url, fmt.Sprintf("container %s", ei.GetName()), ports)
+		if err != nil {
+			return err
+		}
 	}
 	if url.Port == -1 {
```

Comment:
I think we can safely remove this as the check has been moved in the if-else condition above

Reply:
Ah damn missed that :(